linux_os/guide/services/ntp/service_timesyncd_root_distance_configured/oval/shared.xml

```diff
@@ -0,0 +1,52 @@
+<def-group>
+  <definition class="compliance" id="{{{ rule_id }}}" version="1">
+    {{{ oval_metadata("Ensure that timesyncd RootDistanceMaxSec is configured") }}}
+    <criteria comment="Timesyncd servers are configured" operator="AND">
+      <extend_definition comment="service timesyncd enabled"
```

Comment:
I don't think it is necessary to care about service state in this rule. It should be only about parameters values, regardless of the service being enabled or disabled. Could you consider removing it completely in this rule, please? Keeping the rule as simple as possible to ensure the proper values should be ideal. Service state should be managed by other rules.

Reply:
I have two more generic concerns in this area, and would be happy to get your opinion:
- if the rule is checked on its own and fails or reports false positive since the service is not enabled
- or in the case we run all the rules but for some reason, say alphabetic order of execution, this rule reports OK, while there is problem with service enabled, so this rule will report wrongly pass

Another theoretical question, going even further out of context :). not about the rules but remediations. Say we have rule about service enabling and such for service configuration. Both initially failed, so a ansible or bash script is prepared containing remediation for at least those two rules. When that script is executed and say the service enabling is performed first, then configuration, often it is the case that the follow-up check will report pass, but actually the system is not in correct state, since the configuration is not applied, because the service was not restarted for example. What do you think is the best way to approach this scenario?

Reply:
> I have two more generic concerns in this area, and would be happy to get your opinion:
> 
> * if the rule is checked on its own and fails or reports false positive since the service is not enabled

I don't see this situation as a false positive. In general it is better to have more specific and as granular as possible rules. If the rule is checking a configuration value, it will only pass if the configuration value is compliant. Another rule will check if the service is enabled and will only pass if the service is indeed enabled. It is even simpler to maintain the rule description, OVAL and remediation when we have simpler and specific rules. In a context of a profile, where both rules in this example are present, the report will clearly and specifically show what is missing or compliant. That is the idea of a profile, where many rules together creates a context and may complement each other to satisfy a requirement. O don't think it would make so much sense to include only one of these rules in a profile. On the other hand, if we have rules doing more than necessary it could be confusing to realize the reason it is failing.

> * or in the case we run all the rules but for some reason, say alphabetic order of execution, this rule reports OK, while there is problem with service enabled, so this rule will report wrongly pass

There are many cases like this. For example, when a rule is only applicable if a package is present, if the package is not initially installed, the check will report `not applicable` but then the package is installed by another requirement. In a second round the `not applicable` requirement may became a `fail` or `pass`. If it became `fail`, the remediation would be necessary. See https://github.com/OpenSCAP/openscap/issues/1880 for other similar cases. Currently, the simplest solution in this case is to run the scan/remediation once more. Maybe it is not the ideal experience but it is not unreasonable too. Also, it would be a relatively complex case to be solved on the scanner side.

> 
> Another theoretical question, going even further out of context :). not about the rules but remediations. Say we have rule about service enabling and such for service configuration. 

service enabling and service configuration should report `not applicable` if the service package is not present.
Otherwise, both should be checked and, if necessary, remediated.

> Both initially failed, so a ansible or bash script is prepared containing remediation for at least those two rules. When that script is executed and say the service enabling is performed first, then configuration, often it is the case that the follow-up check will report pass, but actually the system is not in correct state, since the configuration is not applied, because the service was not restarted for example. What do you think is the best way to approach this scenario?

If a service needs to be restarted after configuration changes, it would make sense to ensure the service restart or reload in the respective remediation.